tests/gpflow/kernels/test_broadcasting.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2020 the GPflow authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+import numpy as np
+from numpy.testing import assert_allclose
+import pytest
+
+import gpflow
+from gpflow import kernels
+
+
+KERNEL_CLASSES = [
+    # Static
+    kernels.White,
+    kernels.Constant,
+    # Stationary
+    kernels.SquaredExponential,
+    kernels.RationalQuadratic,
+    kernels.Exponential,
+    kernels.Matern12,
+    kernels.Matern32,
+    kernels.Matern52,
+    kernels.Cosine,
+    kernels.Linear,
+    kernels.Polynomial,
+    # Following kernels do not broadcast:
+    pytest.param(kernels.ArcCosine, marks=pytest.mark.xfail),  # broadcasting not implemented
+    pytest.param(kernels.Coregion, marks=pytest.mark.xfail),  # broadcasting not implemented
+    pytest.param(kernels.Periodic, marks=pytest.mark.xfail),  # broadcasting not implemented
+]
+
+
+def _test_no_active_dims(Kernel):
```

Comment:
Could you rename it to `kernel_class`, please? And in other places as well.

Reply:
done